src/provider.coffee

```diff
@@ -6,7 +6,7 @@ extensions        = require './extensions'
 
 
 class Provider
-  constructor: (consumer_key, consumer_secret, nonceStore, signature_method=(new HMAC_SHA1()) ) ->
+  constructor: ({consumer_key, consumer_secret, nonceStore, signature_method=(new HMAC_SHA1()), providerOptions}={}) ->
```

Comment:
I'm afraid we will need to do so (change input into an object). I didn't find another way to pass named params in JS.
So, I will update PR with authEnabled parameter.